src/bucket/Bucket.cpp

```diff
@@ -561,6 +562,22 @@ mergeCasesWithEqualKeys(MergeCounters& mc, BucketInputIterator& oi,
     ++ni;
 }
 
+static void
+checkAllInputHashes(BucketInputIterator& oi, BucketInputIterator& ni,
+                    std::vector<BucketInputIterator>& shadowIterators)
+{
+    oi.checkHash();
+    ni.checkHash();
+    for (auto& shadow : shadowIterators)
```

Comment:
Curious about performance here: how does this affect IO and merge latency? Are we reading much more data, as we're iterating through all shadows during a merge? (this will become irrelevant with CAP-0025, but should be considered if this change is to be released prior to protocol 12)

Reply:
I didn't compare merge latency or IO load yet. Fair to ask! Though I do expect we'll be taking CAP-0025; but it'd be good to know. Will check soon.